### PR TITLE
SALTO-5738 support deploy for page restrictions

### PR DESCRIPTION
**Pull request by netama**
Clean up redundant details on restrictions on fetch + support deploy.

---

Tested on addition + modification

---
_Release Notes_: 
_Confluence Adapter_:
* Add support for deploying page restrictions

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/confluence-adapter/src/definitions/deploy/deploy.ts [hunk 1]
```diff
@@ -50,6 +50,11 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
         customizations: {
           add: [
             {
+              condition: {
+                transformForCheck: {
+                  omit: ['restriction', 'version'],
```
Comment (omrifarkash) [inline packages/confluence-adapter/src/definitions/deploy/deploy.ts:55]: Does it matter on addition change? 
I think in addition this condition will always be true with or without omitting those fields?

File: packages/confluence-adapter/src/definitions/deploy/deploy.ts [hunk 2]
```diff
@@ -60,9 +65,26 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
...
             },
+            {
+              request: {
+                endpoint: {
+                  path: '/wiki/rest/api/content/{id}/restriction',
+                  method: 'put',
+                },
+                transformation: {
+                  pick: ['restriction'],
+                  adjust: ({ value }) => ({ value: { results: _.get(value, 'restriction') } }),
```
Comment (omrifarkash) [inline packages/confluence-adapter/src/definitions/deploy/deploy.ts:76]: `nestUnderField` won't work?

File: packages/confluence-adapter/src/definitions/fetch/fetch.ts [hunk 3]
```diff
@@ -305,15 +305,20 @@ const createCustomizations = (): Record<string, definitions.fetch.InstanceFetchA
...
           root: 'results',
-          adjust: item => ({
-            value: {
-              operation: _.get(item.value, 'operation'),
-              restrictions: {
-                user: _.get(item.value, 'restrictions.user.results'),
-                group: _.get(item.value, 'restrictions.group.results'),
+          adjust: ({ value }) => {
+            const userRestrictions = _.get(value, 'restrictions.user.results')
+            return {
```
Comment (omrifarkash) [inline packages/confluence-adapter/src/definitions/fetch/fetch.ts:310]: Maybe extract the function to transform_utils folder? just to be aligned with the rest of the code in the adapter.
You can open a ticket about it and put it on me 🙏

**Review by omrifarkash — APPROVED** [review]
Thanks for the help!

File: packages/confluence-adapter/src/definitions/deploy/deploy.ts (hunk 1 above)
Comment (netama) [reply packages/confluence-adapter/src/definitions/deploy/deploy.ts:55]: `transformForCheck` only runs on modifications (nothing to compare to on addition)

Comment (netama) [reply packages/confluence-adapter/src/definitions/deploy/deploy.ts:55]: I can remove it

File: packages/confluence-adapter/src/definitions/deploy/deploy.ts (hunk 2 above)
Comment (netama) [reply packages/confluence-adapter/src/definitions/deploy/deploy.ts:76]: no, it converts it to an object instead of an array

**Pull request merged**